### ISSUE TITLE
GCop.Extra architecture

### ISSUE THREAD

**Issue by PaymonK** [Geeksltd/GCop#117]
Currently anything we put inside GCop.Extra requires an Extension update which is not as easy and agile as a NuGet update.

Is it at all possible to change the architecture so that GCop.Extra becomes just a shell to load the full Roslyn project info and then provide that somehow to the normal rules? For example we know that the normal rules cannot access the file system, but can they send http or tcp requests? 

If they can, then we can potentially turn GCop.Extra into a simple workaround that provides the full roslyn data to the normal rules via HTTP or TCP.

**Comment by Foroughi** [comment]
> we know that the normal rules cannot access the file system

do we ?! i don't agree this. 

You have access to files system in both concepts (Vsix or nuggets). The problem of iterating the files inside the project (or solution) was about the performance problem. Also the main problem (that we decided to move back to vsix version)  was that we can not share a data between some rules. Imagine this : 

I read all the files in project (We know its gonna take a while for big projects) then i have to store the data colllected from the files somewhere. where exactly ?! in a file in the root of solution for example ? or  in a database ?! this scenario has been checked by Karvan on Pegasus (according to him , pegasus is not a big project). The build action was taking time almost forever! 

Im not gonna say we can not do this, but im informing you that we are gonna face with performance poblem. Even now that we`ve seperated the wide-solution on an vsix project, we are still having problem with nuggets packages. Last week Karvan told me that you guys have some problem about this on nugget packages !! now imagine if we wanna put more proccess on them !!!

I talked with roslyn team about this. There was an argue between them. They warned me that im gonna face with performance problem and we did.
